### ISSUE TITLE
A non-blocking, cross-platform async executor `SystemParam`

### ISSUE THREAD

**Issue by simbleau** [bevyengine/bevy#8798]
## What problem does this solve or what need does it fill?

To run any async task in the background on Bevy, one must spawn a task on the `AsyncComputeTaskPool` in Bevy and detach it.

This leads one to constantly repeat the boiler plate:
```rs
let task_pool = AsyncComputeTaskPool::get();
let handle = task_pool.spawn(async { 5 });
handle.detach();
```

Also, there's a huge problem here that is **very** difficult to solve for new players to Rust.

- "How do I get the result of my async future?"
- "I want to use an reqwest query and use the result without blocking"

There no way to get the result `5` from this closure easily that cross-compiles for WASM/Native, especially since:
- Native closures/async functions are `Send`
- WASM closures-async functions and most other things are `!Send`

Thus, magic is necessary if you want to use a result from a background task.

## What solution would you like?

A solution I tried that worked was creating two abstractions, one for WASM, one for native. They both use `futures::channels::oneshot` channels to send the result of the future when it is ready.

Minimal example (This would actually be two structs, one with `Send` futures, one `!Send`):

```rs
pub struct MyAbstraction(Pin<Box<dyn Future<Output = ()> + 'static>>, Receiver<T>);

impl MyAbstraction {
    fn new(fut: impl Future<Output = T>) -> Self {
        let (tx, rx) = oneshot::channel();
        let new_fut = async move {
            let result = fut.await;
            _ = tx.send(result);
        };
        Self(Box::pin(new_fut), rx)
    }

    pub fn into_parts(
        self,
    ) -> (
        Pin<Box<dyn Future<Output = ()> + 'static>>,
        AsyncReceiver<T>,
    ) {
        (self.0, self.1)
    }
}
```

With this very simple example, one can:
- `let (fut, rx) = MyAbstraction::new(async { 5}).into_parts()`.
- Spawn the `fut` on the `AsyncComputeTaskPool`, and detach it.
- Wait for `rx` to receive the item.

Ta Da! One can spawn the background task and get the result without blocking on Native/WASM.

Let's go a step further. This isn't ergonomic.

I have a fully working solution I call `bevy-async-task`. It comes with my abstraction (above), `AsyncTask` and `AsyncTaskRunner` (a `SystemParam`) to make running async functions extremely simple in the background.

See my examples: https://github.com/vectorgameexperts/bevy-async-task 
(see my bevy-0.11 branch, as well, which is more up-to-date)

I am actually hoping the async executor system param would be something that should go into Bevy as a new SystemParam.

**Comment by simbleau** [comment]
Closing in favor of opening a PR based on https://github.com/vectorgameexperts/bevy-async-task
